engine/engine.go

```diff
@@ -95,23 +92,47 @@ func NewFromSettings(settings *Settings) (*Engine, error) {
 		gctlog.Infoln(gctlog.Global, "Logger initialised.")
 	}
 
-	b.Settings.ConfigFile = filePath
-	b.Settings.DataDir = settings.DataDir
+	b.Settings.ConfigFile = settings.ConfigFile
+	b.Settings.DataDir = b.Config.GetDataPath()
 	b.Settings.CheckParamInteraction = settings.CheckParamInteraction
 
 	err = utils.AdjustGoMaxProcs(settings.GoMaxProcs)
 	if err != nil {
 		return nil, fmt.Errorf("unable to adjust runtime GOMAXPROCS value. Err: %s", err)
 	}
 
-	ValidateSettings(&b, settings)
+	validateSettings(&b, settings)
 	return &b, nil
 }
 
-// ValidateSettings validates and sets all bot settings
-func ValidateSettings(b *Engine, s *Settings) {
-	flag.Visit(func(f *flag.Flag) { flagSet[f.Name] = true })
+// loadConfigWithSettings creates configuration based on the provided settings
+func loadConfigWithSettings(settings *Settings) (*config.Config, error) {
+	filePath, err := config.GetFilePath(settings.ConfigFile)
+	if err != nil {
+		return nil, err
+	}
+	log.Printf("Loading config file %s..\n", filePath)
+
+	conf := &config.Config{}
```

Comment:
while I know its not deal to have globals we currently have config.Cfg which is read by GetConfig()

https://github.com/thrasher-corp/gocryptotrader/blob/091ffae8df1c858abe374bbc8119165e187f1942/config/config.go#L1818-L1820

While its mostly used in test coverage that the old LoadConfig() method there is a few cases in non-test code that use GetConfig() and return it 

https://github.com/thrasher-corp/gocryptotrader/blob/091ffae8df1c858abe374bbc8119165e187f1942/exchanges/exchange.go#L232-L244

https://github.com/thrasher-corp/gocryptotrader/blob/091ffae8df1c858abe374bbc8119165e187f1942/engine/events.go#L309-L318

Because its no longer set as part of loadConfigWithSettings it will cause some nice _unintended consequences_

Reply:
OK, I switched it because of https://github.com/thrasher-corp/gocryptotrader/pull/549#discussion_r487303039
If it's just for test coverage then I would actually prefer to keep the new struct creation, as the old tests are not changed and new tests cover this function.
Switched back anyway.

Reply:
I am happy for it to be changed just the things mentioned above would also need to be updated because they would cause panics when trying to access

Reply:
Yes, these functions would explicitly need access to the config from the call stack, I suspect the changes are not going to be trivial and will need some conceptual decisions.
Anyway, I think it's best to do the refactoring in a separate change, so we can leave this as is...